chickadee/processes/wps_ca.py

```diff
@@ -0,0 +1,177 @@
+import os
+from pywps import Process, ComplexOutput, LiteralInput, FORMATS
+from pywps.app.Common import Metadata
+from netCDF4 import Dataset
+from rpy2 import robjects
+
+from wps_tools.utils import log_handler
+from wps_tools.io import log_level
+from chickadee.utils import logger, set_r_options, get_package
+from chickadee.io import gcm_file, obs_file, varname, num_cores, end_date
+
+
+class CA(Process):
+    """Constructed Analogue (CA) downscaling algorithm:
+    Starts by spatially aggregating high-resolution gridded observations
+    up to the scale of a GCM. Then it proceeds to bias correcting the GCM
+    based on those observations. Finally, it conducts the search for temporal
+    analogues. This involves taking each timestep in the GCM and searching for
+    the top 30 closest timesteps in the gridded observations. For each of the
+    30 closest "analogue" timesteps, CA records the integer number of the
+    timestep (indices) and a weight for each of the analogues."""
+
+    def __init__(self):
+        self.status_percentage_steps = {
+            "start": 0,
+            "process": 10,
+            "write_files": 80,
+            "build_output": 95,
+            "complete": 100,
+        }
+
+        inputs = [
+            gcm_file,
+            obs_file,
+            varname,
+            num_cores,
+            end_date,
+            LiteralInput(
+                "indices",
+                "Indices File",
+                abstract="File name to store indices of analogue times steps (suffix .txt)",
+                data_type="string",
+            ),
+            LiteralInput(
+                "weights",
+                "Weights File",
+                abstract="File name to store weights of analogues (suffix .txt)",
+                data_type="string",
+            ),
+            log_level,
+        ]
+
+        outputs = [
+            ComplexOutput(
+                "indices_file",
+                "Indices File",
+                abstract="File path with analogue indices",
+                supported_formats=[FORMATS.TEXT],
+            ),
+            ComplexOutput(
+                "weights_file",
+                "Weights File",
+                abstract="File path with analogue weights",
+                supported_formats=[FORMATS.TEXT],
+            ),
+        ]
+
+        super(CA, self).__init__(
+            self._handler,
+            identifier="ca",
+            title="CA",
+            abstract="Constructed Analogue (CA) downscaling algorithm",
+            keywords=["constructed", "analogue", "downscaling"],
+            metadata=[
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
```

Comment:
We are using different metadata here, and I think it would be nice to have consistent metadata throughout every process in Chickadee. Perhaps we could assemble yours and [mine](https://github.com/pacificclimate/chickadee/blob/master/chickadee/processes/wps_CI.py#L35) here.